src/main/java/server/dabatase/daoimpl/PostDAO.java

```diff
@@ -0,0 +1,273 @@
+        post.setDatePosted(rs.getDate("date_modified").toLocalDate().atStartOfDay());
+        post.setMessage((rs.getString("message")));
+        post.setAttID(attID);
+
+        return post;
+    }
+
+    public static boolean deletePostDatabase(int postId) {
+        boolean success = false;
+        try {
+            Connection conn = DBConnection.getConnection();
+            if (existsInPostHashtag(conn, postId)) {
+                String query1 = "DELETE from post_hashtag where post_id=?";
+                PreparedStatement pstmt = conn.prepareStatement(query1);
+                pstmt.setInt(1, postId);
+                pstmt.executeUpdate();
+            }
+
+            String sql = "DELETE from post_info where post_id=?";
+
+            PreparedStatement ps = conn.prepareStatement(sql);
+            ps.setInt(1, postId);
+            ps.executeUpdate();
+
+            success = true;
+
+        } catch (Exception e) {
+            System.err.println("Got an exception! ");
+            System.err.println(e.getMessage());
+        }
+        return success;
+    }
+
+    public static boolean existsInPostHashtag(Connection conn, int postId) {
+        boolean exists = false;
+        try {
+            String sql = "SELECT * from post_hashtag where post_id=?";
+
+            PreparedStatement ps = conn.prepareStatement(sql);
+            ps.setInt(1, postId);
+            ResultSet r = ps.executeQuery();
+
+            if (r.next()) {
+                exists = true;
+            }
+        } catch (Exception e) {
+            System.err.println("Got an exception! ");
+            System.err.println(e.getMessage());
+        }
+
+        return exists;
+    }
+
+    public static Post updatePostDatabase(int postId, String uname, String title, String message) {
+        Post post = new Post();
+
+        LocalDateTime localDate = LocalDateTime.now();
+        java.sql.Date date = java.sql.Date.valueOf(localDate.toLocalDate());
+        try {
+            Connection conn = DBConnection.getConnection();
+            if (existsInPostHashtag(conn, postId)) {
+                String query1 = "DELETE from post_hashtag where post_id=?";
+                PreparedStatement pstmt = conn.prepareStatement(query1);
+                pstmt.setInt(1, postId);
+                pstmt.executeUpdate();
+            }
+            String sql = "UPDATE post_info SET username = ?, title = ?, date_modified = ?, message = ? WHERE post_id = ?";
+            PreparedStatement ps = conn.prepareStatement(sql);
+            ps.setString(1, uname);
+            ps.setString(2, title);
+            ps.setDate(3, date);
+            ps.setString(4, message);
+            ps.setInt(5, postId);
+            ps.executeUpdate();
+
+            post.setPostID(postId);
+            post.setUsername(uname);
+            post.setTitle(title);
+            post.setMessage(message);
+            post.setDateModified(date.toLocalDate().atStartOfDay());
```

Comment:
Update sets the date_modified to the start of the current date
I think its from when we were using LocalDate so you just switch to LocalDateTime

Reply:
I think this also applies to post message

Reply:
Yes we should use LocalDateTime instead